src/lib/buildPrompt.ts

```diff
@@ -49,6 +53,38 @@ export async function buildPrompt({
 		];
 	}
 
+	// section to handle potential files input
+	if (model.multimodal) {
+		messages = await Promise.all(
+			messages.map(async (el) => {
+				let content = el.content;
+
+				if (el.from === "user") {
+					if (el.files && url && fetch && id && el.files?.length > 0) {
```

Comment:
```suggestion
					if (el?.files?.length > 0 && url && fetch && id) {
```
I think we can combine the two `el.files` checks here